build.xml

```diff
@@ -27,22 +47,189 @@
+  
+  <target name="validateRngWithRnc" description="Validate generated RELAXNG with its own schema.">
+    <description>
+      TARGET validateRngWithRnc
+      This target validates the generated RELAX NG file with its own RELAX NG (rnc) schema.
+    </description>
+    <dirname property="ploddDir" file="${inputPlodd}"/>
+    <echo message="Validate ${ploddDir}/${ploddName}.rng with ../Schemas/relaxng.rnc."/>
+    <jing compactsyntax="true" rngfile="${basedir}/Schemas/relaxng.rnc" failonerror="true">
+      <fileset file="${ploddDir}/${ploddName}.rng"/>
+    </jing>
+  </target>
+  
+  <target name="validateSchematronWithRng" description="Validate extracted Schematron with its own schema.">
+    <description>
+      TARGET validateSchematronWithRng
+      This target validates the extracted Schematron file with its own RELAX NG schema.
+    </description>
+    <dirname property="ploddDir" file="${inputPlodd}"/>
+    <echo message="Validate ${ploddDir}/${ploddName}.sch with ../Schemas/schematron.rng."/>
+    <jing rngfile="${basedir}/Schemas/schematron.rng" failonerror="true">
+      <fileset file="${ploddDir}/${ploddName}.sch"/>
+    </jing>
+  </target>
+  
+    <target name="validateSchematronWithSch" description="Validate the extracted Schematron with Schematron.">
+    <description>
+      TARGET validateSchematronWithSch
+      This target validates an extracted Schematron file with a local Schematron schema.
+    </description>
+    <dirname property="ploddDir" file="${inputPlodd}"/>
+    <antcall target="compileSchematron">
+      <param name="schSchemaFile" value="${basedir}/Schemas/schematron.sch"/>
+    </antcall>
+    <echo message="Validate ${ploddDir}/${ploddName}.sch with schematron.sch."/>
+    <echo message="Process ${ploddDir}/${ploddName}.sch with compiled schematron_sch.xslt."/>
+    <!-- Generate the SVRL needed. -->
+    <java failonerror="true" fork="true" classname="net.sf.saxon.Transform" classpath="${saxon}" dir="${basedir}">
+      <arg value="-xsl:${basedir}/Schemas/schematron_sch.xslt"/>
+      <arg value="-s:${ploddDir}/${ploddName}.sch"/>
+      <arg value="-o:${ploddDir}/${ploddName}_sch.svrl"/>
+    </java>
+    <!-- Process with SVRL parser. -->
+    <java failonerror="true" fork="true" classname="net.sf.saxon.Transform" classpath="${saxon}" dir="${basedir}">
+      <arg value="-xsl:${basedir}/XSLT/svrl2msgs.xslt"/>
+      <arg value="-s:${ploddDir}/${ploddName}_sch.svrl"/>
+      <arg value="-o:${ploddDir}/${ploddName}_sch_svrl.txt"/>
+    </java>
+    <!-- Read the output to see if there's anything there. Make it quiet so
+         there's no error message when the file is empty. -->
+    <loadfile property="svrlMessages" srcFile="${ploddDir}/${ploddName}_sch_svrl.txt" quiet="true"/>
+    <!-- Fail if there's anything in the file other than a linebreak. -->
+    <fail message="${ploddDir}/${ploddName}.sch: ${svrlMessages}">
+      <condition>
+        <length length="2" string="${svrlMessages}" when="gt"/>
```

Comment:
We (as a group) think 4 or 5 is better, whether length means bytes or characters.